rotkehlchen/premium/premium.py

```diff
@@ -333,23 +354,100 @@ def _register_new_device(self, device_id: str) -> None:
         - PremiumAuthenticationError: if the queried API returns a 401 error
         """
         log.debug(f'Registering new device {device_id}')
+        if (
+            (system_platform := platform.system()) == 'Linux' and
+            (docker_container := check_docker_container()) is not None
+        ):
+            # in the case of docker where we always run under linux. get the container id
+            system_platform = DOCKER_PLATFORM_KEY
+            device_name = docker_container
+        else:
+            device_name = str(uuid4())  # can be edited later.
+
         try:
             response = self.session.put(
                 url=f'{self.rotki_nest}{(method := "devices")}',
                 json=self.sign(
                     method=method,
                     device_identifier=device_id,
-                    device_name=str(uuid4()),  # can be edited later.
-                    platform=platform.system(),
+                    device_name=device_name,
+                    platform=system_platform,
                 ),
                 timeout=ROTKEHLCHEN_SERVER_TIMEOUT,
             )
         except requests.exceptions.RequestException as e:
             raise RemoteError(f'Failed to register device due to: {e}') from e
 
         if response.status_code in (HTTPStatus.CREATED, HTTPStatus.CONFLICT):
+            # Cache Docker device info on successful registration
+            if system_platform == DOCKER_PLATFORM_KEY:
+                self._set_docker_device_info(device_id, get_system_spec()['rotkehlchen'])
             return None  # device was created or it already exists
 
+        if (
+            response.status_code == HTTPStatus.UNPROCESSABLE_ENTITY and
+            system_platform == DOCKER_PLATFORM_KEY
+        ):
+            # Device limit reached for Docker. Check if it's a version updates
+            if (cached_info := self._get_docker_device_info()) is None:
+                # No cached info, can't auto-update
+                check_response_status_code(
+                    response=response,
+                    status_codes=[HTTPStatus.CREATED],
+                )
+                return
+
+            cached_device_id, cached_version = cached_info
+            current_version = Version(get_system_spec()['rotkehlchen'])
+
+            # If current version is not newer, don't auto-delete
+            if current_version <= Version(cached_version):
```

Comment:
```suggestion
            # If current version is not newer, don't auto-delete
            if Version(current_version := get_system_spec()['rotkehlchen']) <= Version(cached_version):
```
This way you won't need to use `str()` below where you reference `current_version`